app/src/main/res/values/strings.xml

```diff
@@ -228,6 +231,7 @@ If you are overwhelmed by the number of quests, you can always fine-tune which q
 
     <string name="confirmation_authorize_now">"You need to authorize with your OSM user account to publish your answers. Authorize now?"</string>
     <string name="confirmation_authorize_now_note2">You can also do this later on the profile screen.</string>
+    <string name="confirmation_permission_upgrade">"To upload a GPS trace, you need to authorize with your OSM user account to publish traces. This will log you out and ask you to re-login with an internet connection. Authorize now?"</string>
```

Comment:
```suggestion
    <string name="confirmation_permission_upgrade">"To upload a GPS trace, you need to authorize with your OpenStreetMap user account to publish traces. This will log you out of the app and ask you to re-login. Proceed now?"</string>
```

* removed internet connection: it's quite obvious for anyone who uses a smartphone that a login requires an internet connection
* do not duplicate the word "authorize" so much -> replace one with "proceed" (IMHO BTW, even better a button named "Proceed" could be that is indicate that the action is somewhat more critical – after all, you could get kicked out of the app if you carelessly tap on "Ok" or whatever is currently shown as the button name. However, I'd still say it's not _that_ criticial after all, as you can use SC without being logged in, for some time at least.)
*  "logged out of the app" sounds better IMHO as it makes clear where you're logged out of…
* removed the abbrevation OSM (quite a technical slang/term) and replaced by full name

Disclaimer: I'm no native English speaker.

Reply:
I'd more agree with the latter part, an audience that does not know that an OSM login requires and internet connection would hardly know what a "GPS trace" is… :upside_down_face:

Reply:
I like @rugk's suggestion and think it's an improvement.

@goldbattle Feel free to explicitly mention "internet connection" if you think it's necessary; I have no strong opinion on that.